External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -623,6 +624,9 @@ namespace FEXCore::Context {
     InitializeCompiler(Thread);
     InitializeThreadData(Thread);
 
+    Thread->CurrentFrame->State.DeferredSignalRefCount.Store(0);
+    Thread->CurrentFrame->State.DeferredSignalFaultAddress = reinterpret_cast<Core::MoveableNonatomicRefCounter<uint64_t>*>(FEXCore::Allocator::mmap(nullptr, 4096, PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_PRIVATE, -1, 0));
```

Comment:
This is something that actually looks really bad if we use placement new.
Sure the initialization looks fine.
```cpp
Thread->CurrentFrame->State.DeferredSignalFaultAddress = new (FEXCore::Allocator::VirtualAlloc(4096)) Core::NonAtomicRefCounter<uint64_t>;
```

But what do when freeing it? We can't just call `delete` on it, we need to call `VirtualFree` instead. This means that one side is using C++ allocation routines and the other half is still casting it. That looks significantly worse.

The alternative is an even bigger footgun. Adding new/delete operators to the class object, then we can just call new/delete directly. That cleans up the Core.cpp code.

```diff
+       void * operator new(size_t size) {
+         LOGMAN_THROW_A_FMT(size <= 4096, "Size of new placement must be smaller than a page.");
+         return FEXCore::Allocator::VirtualAlloc(4096);
+       }
+
+       void operator delete(void * p) {
+         FEXCore::Allocator::VirtualFree(p, 4096);
+       }
+
```
This looks fine and dandy initially but then this introduces a new weirder problem. If the POD type is allocated as part of a different object its size will be `sizeof(T)`, but if allocated through new then its size is a minimum of 4096.
This object is used both ways, one in CPUState directly, and then another allocated to a page with a pointer in CPUState only.
This is absolutely disgusting and I won't have it.

So while this might not look it at first, this is what peak mental model looks like.

Reply:
Yeah, makes sense. Not a fan of the many unnecessary`reinterpreter_cast`s we have, but using overly complex code to address it for a single instance isn't great either.